### ISSUE TITLE
Regularization (i.e. weight decay)

### ISSUE THREAD

**Issue by kmario23** [tensorflow/nmt#213]
Hi,
  The `BLEU` score of the trained NMT system is very low (~ 12.5) even after training for 1M iterations. I was wondering this is because we don't apply `weight decay` (or L2 regularization). Is there a way to use regularization in the current NMT codebase? If yes, at which point do I have to start?

Thanks!
